core/utils/wasm-builder-runner/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # WASM builder runner
+//!
+//! Since cargo contains millions of bugs when it comes to correct dependency and feature
+//! resolution, we need this little tool. See <https://github.com/rust-lang/cargo/issues/5730> for
+//! more information.
+//!
+//! It will create a project that itself will call `substrate-wasm-builder` to not let any dependency
+//! from `substrate-wasm-builder` influence the main project's dependencies.
+//!
+//! For more information see <https://crates.io/substrate-wasm-builder>
+
+use std::{env, process::Command, fs, path::{PathBuf, Path}};
+
+/// Environment variable that tells us to skip building the WASM binary.
+const SKIP_BUILD_ENV: &str = "SKIP_WASM_BUILD";
+
+/// Environment variable that tells us to create a dummy WASM binary.
+///
+/// This is useful for `cargo check` to speed-up the compilation.
+///
+/// # Caution
+///
+/// Enabling this option will just provide `&[]` as WASM binary.
+const DUMMY_WASM_BINARY_ENV: &str = "BUILD_DUMMY_WASM_BINARY";
+
+/// Replace all backslashes with slashes.
+fn replace_back_slashes<T: ToString>(path: T) -> String {
+	path.to_string().replace("\\", "/")
+}
+
+/// The `wasm-builder` dependency source.
+pub enum WasmBuilderSource {
+	/// The relative path to the source code from the current manifest dir.
+	Path(&'static str),
+	/// The git repository that contains the source code.
+	Git {
+		repo: &'static str,
+		rev: &'static str,
+	},
+	/// Use the given version released on crates.io
+	Crates(&'static str),
+}
+
+impl WasmBuilderSource {
+	/// Convert to a valid cargo source declaration.
+	///
+	/// `absolute_path` - The manifest dir.
+	fn to_cargo_source(&self, manifest_dir: &Path) -> String {
+		match self {
+			WasmBuilderSource::Path(path) => {
+				replace_back_slashes(format!("path = \"{}\"", manifest_dir.join(path).display()))
+			}
+			WasmBuilderSource::Git { repo, rev } => {
+				format!("git = \"{}\", rev=\"{}\"", repo, rev)
+			}
+			WasmBuilderSource::Crates(version) => {
+				format!("version = \"{}\"", version)
+			}
+		}
+	}
+}
+
+/// Build the currently build project as WASM binary.
```

Comment:
Do you mean build a project that is already built in native? If yes, use this:
```suggestion
/// Build the currently built project as WASM binary.
```